compiler-rt/lib/sanitizer_common/sanitizer_atomic_clang.h

```diff
@@ -39,47 +46,79 @@ inline void atomic_signal_fence(memory_order) {
 }
 
 inline void atomic_thread_fence(memory_order) {
+#if SANITIZER_CHEERPWASM
+#else
   __sync_synchronize();
+#endif
 }
 
 template<typename T>
 inline typename T::Type atomic_fetch_add(volatile T *a,
     typename T::Type v, memory_order mo) {
+#if SANITIZER_CHEERPWASM
```

Comment:
for all of these,  why can't we use the `__sync_` functions? iirc they are  stubbed in musl?

Reply:
It resulted in all kinds of errors when I was just starting out with ASan, but it seems to just work now. I think it might have something to do that I'm not manually compiling ASan anymore but using the cmake file. I've updated it to use the `__sync_*` functions